internal/pkg/skuba/addons/cilium_test.go

```diff
@@ -0,0 +1,324 @@
+			},
+		)
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			renderContext := renderContext{
+				addon:  tt.addon,
+				config: tt.config,
+			}
+			got := renderContext.CiliumOperatorImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumOperatorImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumImage(t *testing.T) {
+	type test struct {
+		name   string
+		addon  Addon
+		config AddonConfiguration
+		want   string
+	}
+	tests := []test{}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tests = append(
+			tests,
+			test{
+				name: "render Cilium Image URL when cluster version is " + ver.String(),
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+				want: img.ImageRepository + "/cilium:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+			},
+		)
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			renderContext := renderContext{
+				addon:  tt.addon,
+				config: tt.config,
+			}
+			got := renderContext.CiliumImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_ciliumCallbacks_beforeApply(t *testing.T) {
+	type test struct {
+		name               string
+		addonConfiguration AddonConfiguration
+		skubaConfiguration *skuba.SkubaConfiguration
+		wantErr            bool
+	}
+	tests := []test{}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tests = append(
+			tests,
+			test{
+				name: "Test Cilium callbacks beforeApply when cluster version is " + ver.String(),
+				addonConfiguration: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+				skubaConfiguration: &skuba.SkubaConfiguration{
+					AddonsVersion: kubernetes.AddonsVersion{},
+				},
+				wantErr: true,
+			},
+		)
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			c := ciliumCallbacks{}
+			if err := c.beforeApply(tt.addonConfiguration, tt.skubaConfiguration); (err != nil) != tt.wantErr {
+				t.Errorf("ciliumCallbacks.beforeApply() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_ciliumCallbacks_afterApply(t *testing.T) {
```

Comment:
Now, the cilium's `afterApply` do nothing, just return `nil`.
For me, I cannot think what this test case want to test :thinking: